src/main/scala/strawman/collection/Sorted.scala

```diff
@@ -0,0 +1,37 @@
+package strawman.collection
+
+import strawman.collection.mutable.Builder
+
+import scala.Ordering
+
+/** Base trait for sorted collections */
+trait Sorted[A] extends SortedLike[A, Sorted]
+
+trait SortedLike[A, +C[X] <: Sorted[X]]
+  extends SortedPolyTransforms[A, C] {
+
+  def ordering: Ordering[A]
+
+  def range(from: A, until: A): C[A]
+
+}
+
+/** Polymorphic transformation methods on sorted collections */
+trait SortedPolyTransforms[A, +C[X] <: Sorted[X]] {
```

Comment:
I mean that you wouldn't want to overload `BitSet.map` as `def map[B](f: Int => B)(implicit ev: B <:< Int)` but as `def map(f: Int => Int)`